### ISSUE TITLE
Issue with docker-io package dependancy when creating cluster to AWS

### ISSUE THREAD

**Issue by andrew-jones** [openshift/openshift-ansible#318]
Using the default options for install to AWS (us-east-1, default ami etc) I am getting a dependancy issue with docker 1.6.0-4 being available but 1.6.2 is required.

TASK: [openshift_node | Install OpenShift Node package] *********************** 
failed: [openshift-node-f6e75] => {"changed": true, "rc": 1, "results": ["Resolving Dependencies\n--> Running transaction check\n---> Package openshift-node.x86_64 0:1.0.0-0.git.4227.6ffebda.el7.centos will be installed\n--> Processing Dependency: openshift = 1.0.0-0.git.4227.6ffebda.el7.centos for package: openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64\n--> Processing Dependency: docker-io >= 1.6.2 for package: openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64\n--> Processing Dependency: socat for package: openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64\n--> Processing Dependency: tuned-profiles-openshift-node for package: openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64\n--> Running transaction check\n---> Package openshift.x86_64 0:1.0.0-0.git.4227.6ffebda.el7.centos will be installed\n---> Package openshift-node.x86_64 0:1.0.0-0.git.4227.6ffebda.el7.centos will be installed\n--> Processing Dependency: docker-io >= 1.6.2 for package: openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64\n---> Package socat.x86_64 0:1.7.2.4-1.fc21 will be installed\n---> Package tuned-profiles-openshift-node.x86_64 0:1.0.0-0.git.4227.6ffebda.el7.centos will be installed\n--> Processing Dependency: tuned >= 2.3 for package: tuned-profiles-openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64\n--> Running transaction check\n---> Package openshift-node.x86_64 0:1.0.0-0.git.4227.6ffebda.el7.centos will be installed\n--> Processing Dependency: docker-io >= 1.6.2 for package: openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64\n---> Package tuned.noarch 0:2.4.1-5.fc21 will be installed\n--> Processing Dependency: virt-what for package: tuned-2.4.1-5.fc21.noarch\n--> Processing Dependency: virt-what for package: tuned-2.4.1-5.fc21.noarch\n--> Processing Dependency: python-pyudev for package: tuned-2.4.1-5.fc21.noarch\n--> Processing Dependency: python-decorator for package: tuned-2.4.1-5.fc21.noarch\n--> Processing Dependency: pygobject3-base for package: tuned-2.4.1-5.fc21.noarch\n--> Processing Dependency: kernel-tools for package: tuned-2.4.1-5.fc21.noarch\n--> Processing Dependency: hdparm for package: tuned-2.4.1-5.fc21.noarch\n--> Processing Dependency: ethtool for package: tuned-2.4.1-5.fc21.noarch\n--> Processing Dependency: dbus-python for package: tuned-2.4.1-5.fc21.noarch\n--> Running transaction check\n---> Package dbus-python.x86_64 0:1.2.0-7.fc21 will be installed\n--> Processing Dependency: libdbus-glib-1.so.2()(64bit) for package: dbus-python-1.2.0-7.fc21.x86_64\n---> Package ethtool.x86_64 2:3.14-3.fc21 will be installed\n---> Package hdparm.x86_64 0:9.45-2.fc21 will be installed\n---> Package kernel-tools.x86_64 0:4.0.5-200.fc21 will be installed\n--> Processing Dependency: kernel-tools-libs = 4.0.5-200.fc21 for package: kernel-tools-4.0.5-200.fc21.x86_64\n--> Processing Dependency: libpci.so.3(LIBPCI_3.3)(64bit) for package: kernel-tools-4.0.5-200.fc21.x86_64\n--> Processing Dependency: libpci.so.3(LIBPCI_3.0)(64bit) for package: kernel-tools-4.0.5-200.fc21.x86_64\n--> Processing Dependency: libpci.so.3()(64bit) for package: kernel-tools-4.0.5-200.fc21.x86_64\n--> Processing Dependency: libcpupower.so.0()(64bit) for package: kernel-tools-4.0.5-200.fc21.x86_64\n---> Package openshift-node.x86_64 0:1.0.0-0.git.4227.6ffebda.el7.centos will be installed\n--> Processing Dependency: docker-io >= 1.6.2 for package: openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64\n---> Package pygobject3-base.x86_64 0:3.14.0-1.fc21 will be installed\n--> Processing Dependency: gobject-introspection(x86-64) >= 1.39.0 for package: pygobject3-base-3.14.0-1.fc21.x86_64\n--> Processing Dependency: libgirepository-1.0.so.1()(64bit) for package: pygobject3-base-3.14.0-1.fc21.x86_64\n---> Package python-decorator.noarch 0:3.4.0-5.fc21 will be installed\n---> Package python-pyudev.noarch 0:0.16.1-3.fc21 will be installed\n---> Package virt-what.x86_64 0:1.15-1.fc21 will be installed\n--> Running transaction check\n---> Package dbus-glib.x86_64 0:0.104-1.fc21 will be installed\n---> Package gobject-introspection.x86_64 0:1.42.0-1.fc21 will be installed\n---> Package kernel-tools-libs.x86_64 0:4.0.5-200.fc21 will be installed\n---> Package openshift-node.x86_64 0:1.0.0-0.git.4227.6ffebda.el7.centos will be installed\n--> Processing Dependency: docker-io >= 1.6.2 for package: openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64\n---> Package pciutils-libs.x86_64 0:3.3.0-1.fc21 will be installed\n--> Finished Dependency Resolution\n You could try using --skip-broken to work around the problem\n You could try running: rpm -Va --nofiles --nodigest\n"]}
msg: Error: Package: openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64 (maxamillion-origin-next)
           Requires: docker-io >= 1.6.2
           Available: docker-io-1.3.2-2.fc21.x86_64 (fedora)
               docker-io = 1.3.2-2.fc21
           Available: docker-io-1.6.0-4.git350a636.fc21.x86_64 (updates)
               docker-io = 1.6.0-4.git350a636.fc21

failed: [openshift-node-14b03] => {"changed": true, "rc": 1, "results": ["Resolving Dependencies\n--> Running transaction check\n---> Package openshift-node.x86_64 0:1.0.0-0.git.4227.6ffebda.el7.centos will be installed\n--> Processing Dependency: openshift = 1.0.0-0.git.4227.6ffebda.el7.centos for package: openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64\n--> Processing Dependency: docker-io >= 1.6.2 for package: openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64\n--> Processing Dependency: socat for package: openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64\n--> Processing Dependency: tuned-profiles-openshift-node for package: openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64\n--> Running transaction check\n---> Package openshift.x86_64 0:1.0.0-0.git.4227.6ffebda.el7.centos will be installed\n---> Package openshift-node.x86_64 0:1.0.0-0.git.4227.6ffebda.el7.centos will be installed\n--> Processing Dependency: docker-io >= 1.6.2 for package: openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64\n---> Package socat.x86_64 0:1.7.2.4-1.fc21 will be installed\n---> Package tuned-profiles-openshift-node.x86_64 0:1.0.0-0.git.4227.6ffebda.el7.centos will be installed\n--> Processing Dependency: tuned >= 2.3 for package: tuned-profiles-openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64\n--> Running transaction check\n---> Package openshift-node.x86_64 0:1.0.0-0.git.4227.6ffebda.el7.centos will be installed\n--> Processing Dependency: docker-io >= 1.6.2 for package: openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64\n---> Package tuned.noarch 0:2.4.1-5.fc21 will be installed\n--> Processing Dependency: virt-what for package: tuned-2.4.1-5.fc21.noarch\n--> Processing Dependency: virt-what for package: tuned-2.4.1-5.fc21.noarch\n--> Processing Dependency: python-pyudev for package: tuned-2.4.1-5.fc21.noarch\n--> Processing Dependency: python-decorator for package: tuned-2.4.1-5.fc21.noarch\n--> Processing Dependency: pygobject3-base for package: tuned-2.4.1-5.fc21.noarch\n--> Processing Dependency: kernel-tools for package: tuned-2.4.1-5.fc21.noarch\n--> Processing Dependency: hdparm for package: tuned-2.4.1-5.fc21.noarch\n--> Processing Dependency: ethtool for package: tuned-2.4.1-5.fc21.noarch\n--> Processing Dependency: dbus-python for package: tuned-2.4.1-5.fc21.noarch\n--> Running transaction check\n---> Package dbus-python.x86_64 0:1.2.0-7.fc21 will be installed\n--> Processing Dependency: libdbus-glib-1.so.2()(64bit) for package: dbus-python-1.2.0-7.fc21.x86_64\n---> Package ethtool.x86_64 2:3.14-3.fc21 will be installed\n---> Package hdparm.x86_64 0:9.45-2.fc21 will be installed\n---> Package kernel-tools.x86_64 0:4.0.5-200.fc21 will be installed\n--> Processing Dependency: kernel-tools-libs = 4.0.5-200.fc21 for package: kernel-tools-4.0.5-200.fc21.x86_64\n--> Processing Dependency: libpci.so.3(LIBPCI_3.3)(64bit) for package: kernel-tools-4.0.5-200.fc21.x86_64\n--> Processing Dependency: libpci.so.3(LIBPCI_3.0)(64bit) for package: kernel-tools-4.0.5-200.fc21.x86_64\n--> Processing Dependency: libpci.so.3()(64bit) for package: kernel-tools-4.0.5-200.fc21.x86_64\n--> Processing Dependency: libcpupower.so.0()(64bit) for package: kernel-tools-4.0.5-200.fc21.x86_64\n---> Package openshift-node.x86_64 0:1.0.0-0.git.4227.6ffebda.el7.centos will be installed\n--> Processing Dependency: docker-io >= 1.6.2 for package: openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64\n---> Package pygobject3-base.x86_64 0:3.14.0-1.fc21 will be installed\n--> Processing Dependency: gobject-introspection(x86-64) >= 1.39.0 for package: pygobject3-base-3.14.0-1.fc21.x86_64\n--> Processing Dependency: libgirepository-1.0.so.1()(64bit) for package: pygobject3-base-3.14.0-1.fc21.x86_64\n---> Package python-decorator.noarch 0:3.4.0-5.fc21 will be installed\n---> Package python-pyudev.noarch 0:0.16.1-3.fc21 will be installed\n---> Package virt-what.x86_64 0:1.15-1.fc21 will be installed\n--> Running transaction check\n---> Package dbus-glib.x86_64 0:0.104-1.fc21 will be installed\n---> Package gobject-introspection.x86_64 0:1.42.0-1.fc21 will be installed\n---> Package kernel-tools-libs.x86_64 0:4.0.5-200.fc21 will be installed\n---> Package openshift-node.x86_64 0:1.0.0-0.git.4227.6ffebda.el7.centos will be installed\n--> Processing Dependency: docker-io >= 1.6.2 for package: openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64\n---> Package pciutils-libs.x86_64 0:3.3.0-1.fc21 will be installed\n--> Finished Dependency Resolution\n You could try using --skip-broken to work around the problem\n You could try running: rpm -Va --nofiles --nodigest\n"]}
msg: Error: Package: openshift-node-1.0.0-0.git.4227.6ffebda.el7.centos.x86_64 (maxamillion-origin-next)
           Requires: docker-io >= 1.6.2
           Available: docker-io-1.3.2-2.fc21.x86_64 (fedora)
               docker-io = 1.3.2-2.fc21
           Available: docker-io-1.6.0-4.git350a636.fc21.x86_64 (updates)
               docker-io = 1.6.0-4.git350a636.fc21

FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/home/user/launch.retry

localhost                  : ok=41   changed=6    unreachable=0    failed=0   
openshift-master-2b609     : ok=63   changed=29   unreachable=0    failed=0   
openshift-node-14b03       : ok=40   changed=12   unreachable=0    failed=1   
openshift-node-f6e75       : ok=40   changed=12   unreachable=0    failed=1   

ACTION [create] failed with exit status 512

**Comment by sborenst** [comment]
Looks like you don't have the right version of docker, maybe try to install it manually.

**Comment by sdodson** [comment]
Unfortunately docker-1.6.2 is a hard requirement as Origin 1.0.0 requires it for secrets to work properly on RHEL/Centos hosts.

It looks like it should be available from the atomic-sig http://ask.projectatomic.io/en/question/313/when-will-docker-162-3-be-available-for-centos-atomic-host/

Given that RHEL released it last week hopefully centos will have it available shortly in the base distribution.

**Comment by andrew-jones** [comment]
So it was the line export ec2_ami='ami-307b3658' in the docs that isn't working.  I can't see anywhere in the ansible config where the ec2_ami env var is being used.  So every create attempt was just using fedora 21.  I've change the config and am pulling Centos 7, which gets updated to 7.1 on create and docker 1.6.2 is available. 

I've hit another issue now,

When the ansible scripts are attempting to install openshift-node, docker 1.6.2 is a dependancy.  Which is available in centos 7.1.  But the version of docker-storage-setup installed from the jdetiber-copr repo during cloud init is conflicting with docker 1.6.2.

file /usr/bin/docker-storage-setup from install of docker-1.6.2-14.el7.centos.x86_64 conflicts with file from package docker-storage-setup-0.0.4-2.el7.centos.noarch

Whats happening here?  I see docker-storage-setup 0.0.5 has been released but no RPM's available yet?  Will this even solve the issue?

**Comment by sdodson** [comment]
@andrew-jones docker-1.6.2-5 and newer from centos/fedora/rhel include docker-storage-setup in the docker rpm rather than in a subpackage. I don't see anything in the ansible module that's installing docker-storage-setup so I guess the centos 7 ami you're spawning has it already installed, perhaps remove it?

Sorry I'm not familiar with the ami provisioning.

**Comment by detiber** [comment]
@sdodson It was getting pulled in through the aws playbooks through cloud-config (user-data), I suspect it still is.

**Comment by andrew-jones** [comment]
@sdodson @detiber It is getting installed via cloud config.  I've commented out everyting in playbooks/aws/openshift-cluster/templates/user_data.j2 and this fixed the docker-storage-setup issue.  

The other issue I had was the aws readme suggested I could set the AMI as an env var.  Like,  
export ec2_ami='ami-307b3658'

This doesn't work.  I've hardcoded the Centos 7.0 AMI into playbooks/aws/openshift-cluster/vars.yml
image: ami-96a818fe

Also the public security group isn't defined by default so I've set,
security_groups: [ 'default' ]

**Comment by detiber** [comment]
@andrew-jones It looks like the aws readme has gotten a bit stale. To override the ami id, the correct env variable is ec2_image.

**Comment by pat2man** [comment]
Can we get the default AMI updated to something that can install docker-1.6.2?

**Comment by pat2man** [comment]
There were a few things broken with the AWS deploy. My branch here seems to deploy correctly:
https://github.com/pat2man/openshift-ansible/tree/aws-fixes

**Comment by detiber** [comment]
@pat2man made a few comments on your branch.  Could you create a pull request with your changes?

**Comment by pat2man** [comment]
@detiber I removed the SDN bit. Must have been something else in my config that was causing it not to work. The AMI seems to always be ami-96a818fe so we should be good there. Pull request: https://github.com/openshift/openshift-ansible/pull/381

**Comment by detiber** [comment]
@brenton this issue can be closed.
